File: smarthouse-backend/src/main/java/com/projects/smarthouse_backend/service/UserService.java
package com.projects.smarthouse_backend.service;

import com.projects.smarthouse_backend.model.User;
import com.projects.smarthouse_backend.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public ResponseEntity<User> getUserById(Long id) {
        User user = userRepository.findById(id).orElse(null);
        return user != null ? ResponseEntity.ok(user) : ResponseEntity.notFound().build();
    }

    public User createUser(User user) {
        return userRepository.save(user);
    }

    public ResponseEntity<User> updateUser(Long id, User userDetails) {
        User user = userRepository.findById(id).orElse(null);
        if (user == null) {
            return ResponseEntity.notFound().build();
        }
        user.setUsername(userDetails.getUsername());
        user.setPassword(userDetails.getPassword());
        user.setEmail(userDetails.getEmail());
        return ResponseEntity.ok(userRepository.save(user));
    }

    public ResponseEntity<Void> deleteUser(Long id) {
        User user = userRepository.findById(id).orElse(null);
        if (user == null) {
            return ResponseEntity.notFound().build();
        }
        userRepository.delete(user);
        return ResponseEntity.noContent().build();
    }
}

File: smarthome-frontend/src/service/userService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api';

export const getUsers = () => axios.get(`${API_BASE_URL}/users`);
export const getUserById = (id) => axios.get(`${API_BASE_URL}/users/${id}`);
export const createUser = (user) => axios.post(`${API_BASE_URL}/users`, user);
export const updateUser = (id, user) => axios.put(`${API_BASE_URL}/users/${id}`, user);
export const deleteUser = (id) => axios.delete(`${API_BASE_URL}/users/${id}`);
export const loginUser = (user) => axios.post(`${API_BASE_URL}/users/login`,user);